subsys/bluetooth/mesh/gatt_services.c

```diff
@@ -740,20 +732,9 @@ int bt_mesh_proxy_gatt_disable(void)
 {
 	BT_DBG("");
 
-	if (gatt_svc == MESH_GATT_NONE) {
-		return -EALREADY;
-	}
-
-	if (gatt_svc != MESH_GATT_PROXY) {
-		return -EBUSY;
-	}
-
 	bt_mesh_proxy_gatt_disconnect();
 
-	bt_gatt_service_unregister(&proxy_svc);
-	gatt_svc = MESH_GATT_NONE;
-
-	return 0;
+	return bt_gatt_service_unregister(&proxy_svc);
```

Comment:
Should this still set `client->gatt_type  = MESH_GATT_NONE`, or is this always handled by the disconnected callback?